sparktestingbase/testcase.py

```diff
@@ -14,68 +16,103 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+from pyspark import SparkConf
+from pyspark.sql import SparkSession
 
 """Provides a common test case base for Python Spark tests"""
 
-from utils import quiet_py4j
-from utils import quiet_logs
+from .utils import quiet_py4j
 
 import unittest2
-from pyspark.context import SparkContext
-from pyspark import HiveContext
-
-import os
 
 
 class SparkTestingBaseTestCase(unittest2.TestCase):
-
-    """Basic common test case for Spark. Provides a Spark context as sc.
+    """Basic common test case for Spark. Provides a Spark session as spark.
     For non local mode testing you can either override sparkMaster
-    or set the enviroment property SPARK_MASTER for non-local mode testing."""
-
-    @classmethod
-    def getMaster(cls):
-        return os.getenv('SPARK_MASTER', "local[4]")
+    set the environment property SPARK_MASTER."""
+
+    def get_conf(self):
+        conf = SparkConf()
+        conf.set('spark.sql.session.timeZone', 'UTC')
+        conf.set('spark.sql.shuffle.partitions', '12')
+        return conf
+
+    def create_local_spark_session(self):
+        class_name = self.__class__.__name__
+        spark = SparkSession \
+            .builder \
+            .master("local[4]") \
+            .appName(class_name) \
+            .enableHiveSupport() \
+            .getOrCreate()
+        spark.sparkContext._conf.setAll([("spark.sql.shuffle.partitions", "12"),
+                                         ("spark.sql.session.timeZone", "UTC")])
```

Comment:
Can we not use the config values from `get_conf` instead of repeating them here?

Reply:
fixed